react/AddressForm.tsx

```diff
@@ -1,46 +1,162 @@
-import React from 'react'
-import { Address } from 'vtex.checkout-graphql'
-import { Input } from 'vtex.styleguide'
+import React, { useState } from 'react'
+import { Input, Dropdown, ButtonPlain } from 'vtex.styleguide'
 import { useAddressContext } from 'vtex.address-context/AddressContext'
-import { LineFragment } from './typings/countryRulesTypes'
+import {
+  LineFragment,
+  Fields,
+  OptionsField,
+  Display,
+} from './typings/countryRulesTypes.d'
+import rules from './countries/rules'
+import PlaceDetails from './PlaceDetails'
+import { FormattedMessage, defineMessages } from 'react-intl'
+import NumberOption from './components/NumberOption'
 
-interface Props {
-  address: Address
-  summary: LineFragment[][]
-}
+defineMessages({
+  country: {
+    defaultMessage: '',
+    id: 'place-components.label.country',
+  },
+  state: {
+    defaultMessage: '',
+    id: 'place-components.label.state',
+  },
+  city: {
+    defaultMessage: '',
+    id: 'place-components.label.city',
+  },
+  neighborhood: {
+    defaultMessage: '',
+    id: 'place-components.label.neighborhood',
+  },
+  ['street-road-avenue']: {
+    defaultMessage: '',
+    id: 'place-components.label.street-road-avenue',
+  },
+  ['number-option']: {
+    defaultMessage: '',
+    id: 'place-components.label.number-option',
+  },
+  complement: {
+    defaultMessage: '',
+    id: 'place-components.label.complement',
+  },
+  stateAbbreviation: {
+    defaultMessage: '',
+    id: 'place-components.label.stateAbbreviation',
+  },
+})
+
+const AddressForm: StorefrontFunctionComponent<{}> = () => {
+  const { address, setAddress } = useAddressContext()
+  const [editing, setEditing] = useState<boolean>(false)
+  const { fields, display } = rules[address.country]
+  const summary = display['extended'] as LineFragment[][]
+
+  const getSummaryFields = (summary: LineFragment[][]) => {
+    let summaryFields = new Set()
+    summary.forEach((line: LineFragment[]) => {
+      line.forEach((fragment: LineFragment) => {
+        summaryFields.add(fragment.name)
+      })
+    })
+    return summaryFields
+  }
```

Comment:
What do you think about moving this function outside the component declaration so, that way, the function isn't defined every time the component is rendered or updated?

Reply:
Seems like a good idea since that function doesn't use any state variables :+1: